modules/bgsegm/test/test_backgroundsubtractor_gbh.cpp

```diff
@@ -132,6 +132,19 @@ void CV_BackgroundSubtractorTest::run(int)
 
 }
 
-TEST(VIDEO_BGSUBGMG, accuracy) { CV_BackgroundSubtractorTest test; test.safe_run(); }
+typedef testing::TestWithParam<std::tuple<perf::MatDepth,int>> bgsubgmg_allTypes;
+TEST_P(bgsubgmg_allTypes, accuracy)
+{
+    const int mtype = CV_MAKETYPE(get<0>(GetParam()), get<1>(GetParam()));
+    CV_BackgroundSubtractorTest test;
```

Comment:
Perhaps the whole test can be reworked in modern style since you've already done large part of this task. In order to do this we need to:
1. Remove `CV_BackgroundSubtractorTest` class and move whole `run` function body to the test body (`mtype` becomes local variable)
2. Use `cv::theRNG()` or `TS::ptr()->get_rng()` for `rng`
3. Use proper `EXPECT_/ASSERT_` macros for error checking instead of `CV_Assert` and `ts->set_failed_test_info(...)`. It can be as simple as just using `EXPECT_GE(TS::ptr()->cmpEps2(...), 0)` or slightly more  changed, e.g. `EXPECT_MAT_NEAR(... ,..., ...)` or `EXPECT_LT(norm(..., ..., ...), some_eps)`

Reply:
Thank you for your comment ! I fixed them.